avalon/nuke/pipeline.py

```diff
@@ -367,12 +458,11 @@ def _on_task_changed(*args):
 def viewer_update_and_undo_stop():
     """Lock viewer from updating and stop recording undo steps"""
     try:
-        # nuke = getattr(sys.modules["__main__"], "nuke", None)
-        # lock all connections between nodes
-        # nuke.Root().knob('lock_connections').setValue(1)
-
         # stop active viewer to update any change
-        nuke.activeViewer().stop()
+        try:
+            nuke.activeViewer().stop()
+        except Exception as e:
```

Comment:
Can we make this error more explicit?
```
try:
    viewer = nuke.activeViewer()
except WhatEverErrorNukeRaises as e:
    log.warning("No available active Viewer: `{}`".format(e))
```

Furthermore I'd expect Nuke to return None or raise a specific error when there is no active viewer.
Because currently you might be failing on actually stopping the viewer! Should be something like this maybe?
```python
viewer = nuke.activeViewer()
if viewer:
    viewer.stop()
```